scipy/optimize/tests/test_optimize.py

```diff
@@ -202,6 +202,40 @@ def test_bfgs(self):
                          [0, -5.24885582e-01, 4.87530347e-01]],
                         atol=1e-14, rtol=1e-7)
 
+    def test_bfgs_dfo(self):
+        # Broyden-Fletcher-Goldfarb-Shanno optimization routine
+        if self.use_wrapper:
+            opts = {'maxiter': self.maxiter, 'disp': self.disp,
+                    'return_all': False}
+            res = optimize.minimize(self.func, self.startparams,
+                                    jac=None, method='BFGS', args=(),
+                                    options=opts)
+
+            params, fopt, gopt, Hopt, func_calls, grad_calls, warnflag = (
+                    res['x'], res['fun'], res['jac'], res['hess_inv'],
+                    res['nfev'], res['njev'], res['status'])
+        else:
+            retval = optimize.fmin_bfgs(self.func, self.startparams, fprime=None,
+                                        args=(), maxiter=self.maxiter,
+                                        full_output=True, disp=self.disp,
+                                        retall=False)
+            (params, fopt, gopt, Hopt,
+             func_calls, grad_calls, warnflag) = retval
+
+        assert_allclose(self.func(params), self.func(self.solution),
+                        atol=1e-6)
+        
+        # Ensure that function call counts are 'known good'; these are from
+        # SciPy 1.21.1. Don't allow them to increase.
+        assert_(self.funccalls == 58, self.funccalls) 
```

Comment:
some of the CI failures look related:

```
scipy/optimize/tests/test_optimize.py:230: in test_bfgs_dfo
    assert_(self.funccalls == 58, self.funccalls)
E   AssertionError: 48
```

Reply:
What command did you run to get this error? When I run `python runtests.py -s optimize` all the tests (including this one) pass.

Reply:
Oh I see it in the continuous integration tests...but I'm not sure why the behavior is different there vs on my machine?

Reply:
It's not that uncommon for there to be platform-specific differences, maybe due to floating point behavior, use of a different compiler, BLAS library, etc.

These kind of function call checks are fragily, and in this case getting a lower number of function calls is a good thing. I think changing `== 58` to `<= 58` should be fine.